unit-tests/shared/src/test/scala/org/scalanative/testsuite/javalib/net/DatagramSocketTest.scala

```diff
@@ -0,0 +1,308 @@
+      ds.close()
+    }
+  }
+
+  @Test def getPort(): Unit = {
+    val ds = new DatagramSocket()
+    try {
+      assertEquals(
+        "Expected -1 for remote port as not connected",
+        -1,
+        ds.getPort()
+      );
+      val port = 49152 // any valid port, even if it is unreachable
+      ds.connect(InetAddress.getLocalHost(), port)
+      assertEquals("getPort returned wrong value", port, ds.getPort())
+    } finally {
+      ds.close()
+    }
+  }
+
+  @Test def receiveBufferSize(): Unit = {
+    // This test basically checks that getReceiveBufferSize &
+    // setReceiveBufferSize do not unexpectedly throw and that the former
+    // returns a minimally sane value.
+    //
+    // The Java 8 documentation at URL
+    // https://docs.oracle.com/javase/8/docs/api/java/net/\
+    //     Socket.html#setReceiveBufferSize-int- [sic trailing dash]
+    // describes the argument for setReceiveBufferSize(int) &
+    // setSendBufferSize(int) as a _hint_ to the operating system, _not_
+    // a requirement or demand.  This description is basically unaltered
+    // in Java 10.
+    //
+    // There are a number of reasons the operating system can choose to
+    // ignore the hint. Changing the buffer size, even before a bind() call,
+    // may not be implemented. The buffer size may already be at its
+    // maximum.
+    //
+    // Since, by definition, the OS can ignore the hint, it makes no
+    // sense to set the size, then re-read it and see if it changed.
+    //
+    // The sendBuffersize test refers to this comment.
+    // Please keep both tests synchronized.
+
+    val ds = new DatagramSocket(null)
+    try {
+      val prevValue = ds.getReceiveBufferSize
+      assertTrue(prevValue > 0)
+      ds.setReceiveBufferSize(prevValue + 100)
+    } finally {
+      ds.close()
+    }
+  }
+
+  @Test def sendBufferSize(): Unit = {
+    // This test basically checks that getSendBufferSize &
+    // setSendBufferSize do not unexpectedly throw and that the former
+    // returns a minimally sane value.
+    // See more extensive comments in setBufferSize test.
+
+    val ds = new DatagramSocket(null)
+    try {
+      val prevValue = ds.getSendBufferSize
+      assertTrue(prevValue > 0)
+      ds.setSendBufferSize(prevValue + 100)
+    } finally {
+      ds.close()
+    }
+  }
+
+  @Test def broadcast(): Unit = {
+    val ds = new DatagramSocket(null)
+    try {
+      val prevValue = ds.getBroadcast()
+      assertTrue(prevValue)
+      ds.setBroadcast(!prevValue)
+    } finally {
+      ds.close()
+    }
+  }
+
+  @Test def bind(): Unit = {
+    val ds1 = new DatagramSocket(null)
+    try {
+      val nonLocalAddr =
+        new InetSocketAddress(InetAddress.getByName("123.123.123.123"), 0)
+      assertThrows("ds1", classOf[BindException], ds1.bind(nonLocalAddr))
+    } finally {
+      ds1.close()
+    }
+
+    val ds2 = new DatagramSocket(null)
+    try {
+      ds2.bind(new InetSocketAddress(InetAddress.getLoopbackAddress, 0))
```

Comment:
`InetAddress.getLoopbackAddress` can probably be done
once (for this test? for entire class?).  It is not going to 
change (unless one passes it to `peek` somewhere).